ostd/src/task/atomic_mode.rs

```diff
@@ -0,0 +1,30 @@
+// SPDX-License-Identifier: MPL-2.0
+
+//! Atomic mode.
+//!
+//! Kernel is in atomic mode when it
+//! * runs in an IRQ handler,
+//! * keeps a [`crate::sync::SpinLock`],
+//! * explicitly disables preemption or IRQ.
+//!
+//! Kernel in atomic mode is supposed to run atomically (i.e., no preemption
+//! or blocking operation is allowed).
+
+#[cfg(debug_assertions)]
+use core::sync::atomic::Ordering;
+
+/// Indicates following blocking operations and panics if in atomic mode.
+pub fn might_break() {
```

Comment:
Now that there is not `AtomicModeGuard` and the Rust doc emphasizes "sleep-like" behaviors, I think this function should be renamed to `might_sleep`.